sdk/download.go

```diff
@@ -151,3 +154,96 @@ func DownloadSDK(version string) error {
 	tarURL := TarballURL(version)
 	return DownloadSignedFile(tarFile, tarURL)
 }
+
+func fileSum(file string) ([]byte, error) {
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	hash := sha512.New()
+	if _, err := io.Copy(hash, f); err != nil {
+		return nil, err
+	}
+
+	return hash.Sum(nil), nil
+}
+
+func compareFileBytes(fileName1, fileName2 string) (bool, error) {
+	sum1, err := fileSum(fileName1)
```

Comment:
why are we checksumming both files here?

the checksum must read in each file entirely before it can be compared

why not compare the contents of the two and abort on the first difference?

Also, why not start this whole thing out with two stat calls which compare the sizes to see if they differ before doing any further IO?

the comparison loop should just be reading a block from file1, block from file2, compare the block, return false on first different block.

fall-through returns true

checksumming is just adding computational overhead, serializing the process, and requiring both files be read entirely always.